app/javascript/crayons/MultiSelectAutocomplete/MultiSelectAutocomplete.jsx

```diff
@@ -0,0 +1,317 @@
+/* eslint-disable jsx-a11y/interactive-supports-focus, jsx-a11y/role-has-required-aria-props */
+// Disabled due to the linter being out of date for combobox role: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/789
+import { h, Fragment } from 'preact';
+import { useState, useEffect, useRef } from 'preact/hooks';
+import { Icon, Button } from '@crayons';
+import { Close } from '@images/x.svg';
+
+const KEYS = {
+  UP: 'ArrowUp',
+  DOWN: 'ArrowDown',
+  ENTER: 'Enter',
+  ESCAPE: 'Escape',
+  DELETE: 'Backspace',
+  COMMA: ',',
+  SPACE: ' ',
+};
+
+const ALLOWED_CHARS_REGEX = /([a-zA-Z0-9])/;
+
+export const MultiSelectAutocomplete = ({ labelText, fetchSuggestions }) => {
+  const [suggestions, setSuggestions] = useState([]);
+  const [selectedItems, setSelectedItems] = useState([]);
+  const [inputPosition, setInputPosition] = useState(null);
+  const [editValue, setEditValue] = useState('');
+  const [activeDescendentIndex, setActiveDescendentIndex] = useState(null);
+  const [ignoreBlur, setIgnoreBlur] = useState(false);
+
+  const inputRef = useRef(null);
+  const inputSizerRef = useRef(null);
+  const selectedItemsRef = useRef(null);
+
+  const handleInputBlur = () => {
+    // Since the input is sometimes removed and rendered in a new location on blur, it's possible that inputRef.current may be null when we complete this check.
+    const currentValue = inputRef.current ? inputRef.current.value : '';
+    // The input will blur when user selects an option from the dropdown via mouse click. The ignoreBlur boolean lets us know we can ignore this event.
+    if (!ignoreBlur && currentValue !== '') {
+      selectItem({ selectedItem: currentValue, focusInput: false });
+    }
+    setIgnoreBlur(false);
+  };
+
+  useEffect(() => {
+    if (inputPosition !== null) {
+      resizeInputToContentSize();
+
+      const { current: input } = inputRef;
+      input.value = editValue;
+      const { length: cursorPosition } = editValue;
+      input.focus();
+      input.setSelectionRange(cursorPosition, cursorPosition);
+    } else {
+      // Remove inline style added to size the input
+      inputRef.current.style.width = '';
+      inputRef.current.focus();
+    }
+  }, [inputPosition, editValue]);
```

Comment:
Refactor (non-blocking): The if and else both call the input's focus().

```suggestion
  useEffect(() => {
    const { current: input } = inputRef;

    if (inputPosition !== null) {
      resizeInputToContentSize();

      input.value = editValue;
      const { length: cursorPosition } = editValue;
      input.setSelectionRange(cursorPosition, cursorPosition);
    } else {
      // Remove inline style added to size the input
      input.style.width = '';
    }

    input.focus();
  }, [inputPosition, editValue]);
```

Reply:
Hmm it's not quite so easy because I need to set the focus before `input.setSelectionRange(cursorPosition, cursorPosition);` in the `true` case. I could do another `if (inputPosition !== null)` check after `input.focus()` but I think maybe it's cleaner with just the one if/else.

I'll definitely bring the `const { current: input } = inputRef;` out of the if/else though as both cases need access to the input

Reply:
This one I'm pretty sure the selection was still made even though the focus came after, but it was near the end of my day.😅